src/main/java/org/jenkinsci/plugins/envinject/service/EnvInjectMasterEnvVarsSetter.java

```diff
@@ -24,22 +32,23 @@ public EnvInjectMasterEnvVarsSetter(@Nonnull EnvVars enVars) {
     @Override
     public Void call() throws EnvInjectException {
         try {
-            Field platformField = EnvVars.class.getDeclaredField("platform");
-            platformField.setAccessible(true);
-            platformField.set(enVars, Platform.current());
+            if (Jenkins.getVersion().isOlderThan(jenkinsVersionWithSetPlatform)) {
+                Field platformField = EnvVars.class.getDeclaredField("platform");
+
+                platformField.setAccessible(true);
+                platformField.set(enVars, Platform.current());                
+            } else {
+                Method method = EnvVars.class.getDeclaredMethod("setPlatform", Platform.class);
+                method.invoke(enVars, Platform.current());
+            }
             if (Main.isUnitTest || Main.isDevelopmentMode) {
                 enVars.remove("MAVEN_OPTS");
             }
-            Field masterEnvVarsFiled = EnvVars.class.getDeclaredField("masterEnvVars");
-            masterEnvVarsFiled.setAccessible(true);
-            Field modifiersField = Field.class.getDeclaredField("modifiers");
-            modifiersField.setAccessible(true);
-            modifiersField.setInt(masterEnvVarsFiled, masterEnvVarsFiled.getModifiers() & ~Modifier.FINAL);
-            masterEnvVarsFiled.set(null, enVars);
-        } catch (IllegalAccessException iae) {
-            throw new EnvInjectException(iae);
-        } catch (NoSuchFieldException nsfe) {
-            throw new EnvInjectException(nsfe);
+            
+            EnvVars.masterEnvVars.clear();
+            EnvVars.masterEnvVars.putAll(enVars); 
```

Comment:
It is generally a terrible idea in Jenkins introduced before 2008.
The implementation here actually causes a concurrency risk since `masterEnvVars` is not synchronized. It may lead to concurrent modification exception
I would suggest replacing one collection by another instance instead of `clear()`=>`putAll()`